src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesConsumer.java

```diff
@@ -77,14 +77,15 @@ public void addBinaryField(FieldInfo field, DocValuesProducer valuesProducer) th
     }
 
     public void addKNNBinaryField(FieldInfo field, DocValuesProducer valuesProducer) throws IOException {
-
         // Get values to be indexed
         BinaryDocValues values = valuesProducer.getBinary(field);
         KNNCodecUtil.Pair pair = KNNCodecUtil.getFloats(values);
         if (pair.vectors.length == 0 || pair.docs.length == 0) {
             logger.info("Skipping engine index creation as there are no vectors or docs in the documents");
             return;
         }
+        // for Lucene engine the creation of index is done by code delegate, no need in additional native data structures
+        if (isLuceneEngine(field)) return;
```

Comment:
for initial implementation adding this logic to existing doc format class, we may need to refactor it to a new format class for 920 codec